src/app/modules/users/components/users-list/users-list.component.ts

```diff
@@ -73,4 +105,18 @@ export class UsersListComponent implements OnInit, OnDestroy, AfterViewInit {
       ? this.store.dispatch(new RevokeRoleUser(userId, roleId))
       : this.store.dispatch(new GrantRoleUser(userId, roleId));
   }
+
+  switchGroup(userId: string, userGroups: string[], groupName: string) {
+    userGroups.includes(groupName)
+      ? this.store.dispatch(new RemoveUserFromGroup(userId, groupName))
+      : this.store.dispatch(new AddUserToGroup(userId, groupName));
+  }
```

Comment:
- It's a good idea to add a return type of function.
- Can you test if this way is more convenient and does not break the tests.

```javascript
switchGroup(userId: string, userGroups: string[], groupName: string) {
    this.store.dispatch(
        userGroups.includes(groupName)
             ? new RemoveUserFromGroup(userId, groupName)
             : new AddUserToGroup(userId, groupName)
    );
}
```
- I think that the `userId `and `userGroups `parameters could be unified, as they are retrieved from the same `User `object. 
- You could put the `groupName `as the first parameter, so the function becomes easier to read.
- Imagine this:
   - switchGroup
   - What group? time-tracker-admin
   - Does who? user